src/main/scala/com/github/sstone/amqp/ChannelOwner.scala

```diff
@@ -185,13 +185,13 @@ class ChannelOwner(init: Seq[Request] = Seq.empty[Request], channelParams: Optio
       forwarder ! AddShutdownListener(self)
       forwarder ! AddReturnListener(self)
       onChannel(channel, forwarder)
-      requestLog.map(r => self forward r)
+      requestLog.foreach { case (request, sender) => self.tell(request, sender.getOrElse(context.sender())) }
       log.info(s"got channel $channel")
       statusListeners.map(a => a ! Connected)
       context.become(connected(channel, forwarder))
     }
     case Record(request: Request) => {
-      requestLog :+= request
+      requestLog :+= request -> Some(sender())
```

Comment:
When we get requests while disconnected, save the sender so that we reply to them later